### DISCUSSION TITLE
Wrong Sensor Values

### DISCUSSION THREAD

**Discussion by vbueschken** [john30/ebusd#1014] [Q&A]
Im using the newest Version of the ebusd adapter and hav integrated it to HA with the AddOn from Lukas Grebe.
I got tonns of Sensors now in Home assistant.
Today I tried to build a dashboard for energie consumption, but the sensor values from ebus have no relation to the values I can read from the 700  display of my heating - total different values.

Something wrong with my Integration? Is it the wrong way to use the AddOn? What's best practice?

**Comment by pm90pl** [comment]
Hi there - what registers do you read and from what circuits/devices? Could you provide an example?

**Reply by vbueschken** [reply]
Sorry @pm90pl I#m an newbie and dummie ....
I have just installed the Hardware (ebusd adapter) and the AddOn from Lukas Grebe.

Then I tried to identify identify / match the Sensor values I found in the automatic created entities campared to the values (energie consumption) I can access direcly on the VRC 700 Terminal at my heating. It was not possible to find a match because of totaly different values.
So sorry - I have no idea what you mean with "..what registers do you read.."

**Reply by pm90pl** [reply]
Any screenshots of entities from HA?

**Reply by vbueschken** [reply]
I got tonns of entities:
here a List from the MQTT Explorer:
ebusd
global
version = "ebusd 23.2.p20230716"
running = true
updatecheck = "revision 23.2 available"
scan = "finished"
signal = true
uptime = 96
bai
SetMode = { "hcmode": {"value": "auto"}, "flowtempdesired": {"value": 0.0}, "hwctempdesired": {"value": null}, "hwcflowtempdesired": {"value": null}, "disablehc": {"value": 1}, "disablehwctapping": {"value": 0}, "disablehwcload": {"value": 0}, "remoteControlHcPump": {"value": 0}, "releaseBackup": {"value": 0}, "releaseCooling": {"value": 0}}
Status01 = { "0": {"name": "temp1", "value": 36.5}, "1": {"name": "temp1", "value": 35.0}, "2": {"name": "temp2", "value": 23.875}, "3": {"name": "temp1", "value": 34.5}, "4": {"name": "temp1", "value": 48.5}, "5": {"name": "pumpstate", "value": "off"}}
HwcStarts = { "0": {"name": "", "value": 800}}
HwcTemp = { "temp": {"value": 35.31}, "sensor": {"value": "ok"}}
HwcTempMax = { "temp": {"value": 65.00}}
DateTime = { "dcfstate": {"value": "valid"}, "btime": {"value": "17:20:14"}, "bdate": {"value": "21.09.2023"}, "temp2": {"value": 23.875}}
HwcUnderHundredStarts = { "0": {"name": "", "value": 4}}
StatusCirPump = { "0": {"name": "", "value": "on"}}
Status02 = { "0": {"name": "hwcmode", "value": "auto"}, "1": {"name": "temp0", "value": 60}, "2": {"name": "temp1", "value": 75.0}, "3": {"name": "temp0", "value": 70}, "4": {"name": "temp1", "value": 65.0}}
maintenancedata_HwcTempMax = { "temp": {"value": 57.44}}
ModulationTempDesired = { "0": {"name": "", "value": 22.1}}
OutdoorstempSensor = { "temp": {"value": 24.25}, "sensor": {"value": "ok"}}
OverflowCounter = { "yesno": {"value": "yes"}}
PowerValue = { "0": {"name": "", "value": "13 03 58 0e 64 10"}}
PrAPSCounter = { "0": {"name": "", "value": 0}}
PrEnergyCountHc1 = { "0": {"name": "", "value": 8141585}}
PrEnergyCountHc2 = { "0": {"name": "", "value": 0}}
PrEnergyCountHc3 = { "0": {"name": "", "value": 0}}
PrEnergyCountHwc1 = { "0": {"name": "", "value": 443554}}
PrEnergyCountHwc2 = { "0": {"name": "", "value": 0}}
PrEnergyCountHwc3 = { "0": {"name": "", "value": 0}}
PrEnergySumHc1 = { "0": {"name": "", "value": 967381520}}
PrEnergySumHc2 = { "0": {"name": "", "value": 0}}
PrEnergySumHc3 = { "0": {"name": "", "value": 0}}
PrEnergySumHwc1 = { "0": {"name": "", "value": 105474363}}
PrEnergySumHwc2 = { "0": {"name": "", "value": 0}}
PrEnergySumHwc3 = { "0": {"name": "", "value": 0}}
PrVortexFlowSensorValue = { "0": {"name": "", "value": 0}}
PumpHours = { "hoursum2": {"value": 3174}}
PumpPower = { "0": {"name": "", "value": 0}}
PumpPowerDesired = { "0": {"name": "", "value": "auto"}}
ReturnTemp = { "temp": {"value": 36.25}, "tempmirror": {"value": 64955}, "sensor": {"value": "ok"}}
ReturnTempExternal = { "temp": {"value": -1.81}, "sensor": {"value": "cutoff"}}
ReturnTempMax = { "temp": {"value": 73.62}}
SHEMaxFlowTemp = { "temp": {"value": 0.00}}
StorageExitTemp = { "temp": {"value": 42.56}, "sensor": {"value": "ok"}}
StorageLoadPumpHours = { "hoursum2": {"value": 132}}
StorageloadPumpStarts = { "cntstarts2": {"value": 1033}}
StorageTemp = { "temp": {"value": 48.69}, "sensor": {"value": "ok"}}
StorageTempDesired = { "temp": {"value": 50.00}}
StorageTempMax = { "temp": {"value": 50.69}}
TempDiffBlock = { "temp0": {"value": 0}}
TempDiffFailure = { "temp0": {"value": 0}}
TempGradientFailure = { "temp0": {"value": 0}}
Templimiter = { "0": {"name": "", "value": "off"}}
TemplimiterWithNTC = { "0": {"name": "", "value": "yes"}}
TempMaxDiffExtTFT = { "temp": {"value": 0.00}}
ValveStarts = { "cntstarts2": {"value": 1793}}
WaterpressureMeasureCounter = { "0": {"name": "", "value": 226}}
WPPostrunTime = { "minutes0": {"value": 5}}
CounterStartattempts1 = { "temp0": {"value": 0}}
CounterStartattempts2 = { "temp0": {"value": 0}}
CounterStartAttempts3 = { "temp0": {"value": 0}}
CounterStartAttempts4 = { "temp0": {"value": 0}}
DeactivationsTemplimiter = { "0": {"name": "", "value": 0}}
DisplayMode = { "0": {"name": "", "value": 2}}
ExtFlowTempDesiredMin = { "temp": {"value": 40.00}}
FanHours = { "hoursum2": {"value": 2402}}
FanStarts = { "cntstarts2": {"value": 0}}
FlowTemp = { "temp": {"value": 36.81}, "sensor": {"value": "ok"}}
FlowTempDesired = { "temp": {"value": 0.00}}
FlowTempMax = { "temp": {"value": 83.75}}
HcHours = { "hoursum2": {"value": 2258}}
HcPumpMode = { "0": {"name": "", "value": 3}}
HcPumpStarts = { "cntstarts2": {"value": 9485}}
HcStarts = { "0": {"name": "", "value": 8700}}
HcUnderHundredStarts = { "0": {"name": "", "value": 9}}
HoursTillService = { "hoursum2": {"value": null}}
HwcHours = { "hoursum2": {"value": 124}}
broadcast
outsidetemp = { "temp2": {"value": 24.062}}
vdatetime = { "time": {"value": "17:20:13"}, "date": {"value": "21.09.2023"}}
700
AdaptHeatCurve = { "yesno": {"value": "no"}}
DisplayedOutsideTemp = { "tempv": {"value": 24.0625}}
Hc1ActualFlowTempDesired = { "tempv": {"value": 0.0}}
Hc1AutoOffMode = { "offmode": {"value": "eco"}}
Hc1ExcessTemp = { "calibrationv": {"value": 0.0}}
Hc1FlowTemp = { "tempv": {"value": 37.5}}
Hc1HeatCurve = { "0": {"name": "", "value": 1.4}}
Hc1HeatCurveAdaption = { "0": {"name": "", "value": 0.0}}
Hc1MaxFlowTempDesired = { "tempv": {"value": 75}}
Hc1MinFlowTempDesired = { "tempv": {"value": 15}}
Hc1PumpStatus = { "0": {"name": "", "value": 0}}
Hc1RoomTempSwitchOn = { "rcmode": {"value": "off"}}
Hc1Status = { "0": {"name": "", "value": 0}}
Hc1SummerTempLimit = { "tempv": {"value": 19}}
Hc2ActualFlowTempDesired = { "tempv": {"value": 0.0}}
Hc2AutoOffMode = { "offmode": {"value": "eco"}}
Hc2ExcessTemp = { "calibrationv": {"value": 0.0}}
Hc2FlowTemp = { "tempv": {"value": null}}
Hc2HeatCurve = { "0": {"name": "", "value": 1.2}}
Hc2HeatCurveAdaption = { "0": {"name": "", "value": 0.0}}
Hc2MaxFlowTempDesired = { "tempv": {"value": 90}}
Hc2MinFlowTempDesired = { "tempv": {"value": 15}}
Hc2PumpStatus = { "0": {"name": "", "value": 0}}
Hc2RoomTempSwitchOn = { "rcmode": {"value": "off"}}
Hc2Status = { "0": {"name": "", "value": 0}}
Hc2SummerTempLimit = { "tempv": {"value": 21}}
Hc3ActualFlowTempDesired = { "tempv": {"value": 0.0}}
Hc3AutoOffMode = { "offmode": {"value": "eco"}}
Hc3ExcessTemp = { "calibrationv": {"value": 0.0}}
Hc3FlowTemp = { "tempv": {"value": null}}
Hc3HeatCurve = { "0": {"name": "", "value": 1.2}}
Hc3HeatCurveAdaption = { "0": {"name": "", "value": 0.0}}
Hc3MaxFlowTempDesired = { "tempv": {"value": 90}}
Hc3MinFlowTempDesired = { "tempv": {"value": 15}}
Hc3PumpStatus = { "0": {"name": "", "value": 0}}
Hc3RoomTempSwitchOn = { "rcmode": {"value": "off"}}
Hc3Status = { "0": {"name": "", "value": 0}}
Hc3SummerTempLimit = { "tempv": {"value": 21}}
HcStorageTempTop = { "tempv": {"value": null}}
HolidayTemp = { "tempv": {"value": 18}}
HwcFlowTemp = { "tempv": {"value": 0.0}}
HwcMaxFlowTempDesired = { "tempv": {"value": 80}}
HwcOpMode = { "opmode": {"value": "auto"}}
HwcSFMode = { "sfmode": {"value": "auto"}}
HwcStorageTemp = { "tempv": {"value": 48.5}}
HwcStorageTempBottom = { "tempv": {"value": null}}
HwcStorageTempTop = { "tempv": {"value": null}}
OpMode = { "opmode": {"value": "auto"}}
OpModeCooling = { "opmode": {"value": "auto"}}
OpModeEffect = { "zonesel": {"value": "alle"}}
OpModeVentilation = { "opmode": {"value": "auto"}}
OutsideTempAvg = { "tempv": {"value": 18.4375}}
PrEnergySum = { "energy4": {"value": 148}}
PrEnergySumHc = { "energy4": {"value": 135}}
PrEnergySumHcLastMonth = { "energy4": {"value": 3}}
PrEnergySumHcThisMonth = { "energy4": {"value": 1}}
PrEnergySumHwcLastMonth = { "energy4": {"value": 0}}
PrEnergySumHwcThisMonth = { "energy4": {"value": 0}}
SFMode = { "sfmode": {"value": "auto"}}
SolarYieldTotal = { "energy4": {"value": 0}}
SystemFlowTemp = { "tempv": {"value": null}}
YieldTotal = { "energy4": {"value": 0}}
z1ActualRoomTempDesired = { "tempv": {"value": 0.0}}
z1CoolingTemp = { "tempv": {"value": 24}}
z1DayTemp = { "tempv": {"value": 24}}
z1HolidayTemp = { "tempv": {"value": 18}}
z1NightTemp = { "tempv": {"value": 17}}
z1OpMode = { "opmode": {"value": "auto"}}
z1OpModeCooling = { "opmode": {"value": "auto"}}
z1QuickVetoTemp = { "tempv": {"value": 24}}
z1RoomTemp = { "tempv": {"value": null}}
z1SFMode = { "sfmode": {"value": "auto"}}
z1ValveStatus = { "0": {"name": "", "value": 0}}
z2ActualRoomTempDesired = { "tempv": {"value": 0.0}}
z2CoolingTemp = { "tempv": {"value": 24}}
z2DayTemp = { "tempv": {"value": 20}}
HcStorageTempBottom = { "tempv": {"value": null}}
HwcTempDesired = { "tempv": {"value": 50}}
PrEnergySumHwc = { "energy4": {"value": 13}}
z2HolidayTemp = { "tempv": {"value": 18}}

**Comment by pm90pl** [comment]
And what values can you find in VRC700?

**Reply by vbueschken** [reply]
![700](https://github.com/john30/ebusd/assets/138365084/f5884cf5-c86c-46bc-875a-bc4cd89b8481)
One example:
Last Month
Heating
electric energy consumption: 3KWh
gas consumption: 51KWh

**Reply by pm90pl** [reply]
look at the value PrEnergyCountHwc1; it seems to be in hecto calories; when you convert it to kWh -> 443554 * 100 = 44355400 calories; it is more/less 51.550831556 kWh -> so it might be a counter for 'Last month hot water gas usage'

**Reply by vbueschken** [reply]
May be you are right... i will try to convert all other values to see lf it matches

**Reply by pm90pl** [reply]
Sure, let me know

**Comment by vbueschken** [comment]
What is confusing for me:
the sensors /entities i got are not the same as in this csv file:
[https://github.com/john30/ebusd-configuration/blob/master/ebusd-2.1.x/de/vaillant/15.700.csv](url)

**Reply by pm90pl** [reply]
Shouldn't - not all registres are exposed to MQTT. One value regarding electricity consumption seems to be right. The gas usage yeah, can't find it.

**Reply by vbueschken** [reply]
Agree, I also couldnt find it.. like several other values. :-(

**Comment by vbueschken** [comment]
No, doesn’t work … ☹

 

Looks like all Sensors for electricity are available with the right values – for gas I couldn’t find a calculation factor that works

 

Von: pm90pl ***@***.***> 
Gesendet: Samstag, 23. September 2023 09:39
An: john30/ebusd ***@***.***>
Cc: Volker ***@***.***>; Author ***@***.***>
Betreff: Re: [john30/ebusd] Wrong Sensor Values (Discussion #1014)

 

Sure, let me know

**Comment by vbueschken** [comment]
I wrote an email to Vaillant technical support and asked for information.
I don't have much hope that I will get a qualified answer there. 
Lets give it a try

**Comment by vbueschken** [comment]
found this:
[https://community.home-assistant.io/t/deriving-gas-consumption-from-ebus-vaillant-ecotec-energy-counters/378358](url)
